api/controllers/WebActionsController.js

```diff
@@ -127,6 +128,77 @@ module.exports = {
     }
   },
 
+  splash: function(req, res) {
+    let { EMAIL, FNAME, PHONE, group } = req.body;
+    let memberHash = crypto
+      .createHash('md5')
+      .update(EMAIL.toLowerCase())
+      .digest('hex');
+    let redirectUrl = '/confirmation';
+
+    const mailchimpUpdateRequest = {
+      method: 'PATCH',
+      uri: `${sails.config.globals.mailchimpApiUrl}/lists/${sails.config
+        .globals.mailchimpListId}/members/${memberHash}`,
+      json: true,
+      auth: {
+        user: sails.config.globals.mailchimpApiAuthUser,
+        pass: sails.config.globals.mailchimpApiAuthPass,
+      },
+      body: {
+        status: 'subscribed',
+        // Subscribe users to Shine Splash App Group
+        interests: { e8db15c44a: true },
+      },
+    };
+
+    const mailchimpSubscribeRequest = {
+      method: 'POST',
+      uri: `${sails.config.globals.mailchimpApiUrl}/lists/${sails.config.globals
+        .mailchimpListId}/members`,
+      json: true,
+      auth: {
+        user: sails.config.globals.mailchimpApiAuthUser,
+        pass: sails.config.globals.mailchimpApiAuthPass,
+      },
+      body: {
+        email_address: EMAIL,
+        status: 'subscribed',
+        merge_fields: {
+          FNAME: FNAME,
+          PHONE: PHONE,
+        },
+        interests: { e8db15c44a: true }, // Subscribe users to Shine Splash App Group
+      },
+    };
+
+    request
+      .patchAsync(mailchimpUpdateRequest)
+      .then(response => {
+        if (!response || !response.body) {
+          sails.log.error(
+            'Invalid response received from MailChimp subscribe call.'
+          );
+        } else if (response.body.status === 404) {
+          request.postAsync(mailchimpSubscribeRequest).then(response => {
```

Comment:
This is fine for now, but it does start to veer into the territory of callback hell. It's also kinda iffy that we're reusing the `response` var name when the previously defined one would still be in scope. It could cause for some confusion with any future changes we might make here.

If it seems like it'd be simple enough, you could try to `return request.postAsync()` which would give you a Promise and then add another `.then` to handle its response.

Or, since bluebird is available already here, you could try rewriting this within a `Promise.coroutine` so it reads more like synchronous code.